controllers/diet_requirement.py

```diff
@@ -84,6 +85,7 @@ def post(self):
     This is a class to retrieve the dietary requirement data from the database
     """
     @requires_auth
+    @is_user_or_has_role('id', UserType.VOLUNTEER, UserType.ROOT_ADMIN)
```

Comment:
Is this correct? The ID parameter should be the user ID. It looks like all these are just the ID of the thing